src/codewhisperer/service/keyStrokeHandler.ts

```diff
@@ -42,6 +42,28 @@ export class KeyStrokeHandler {
         return (this.#instance ??= new this())
     }
 
+    private startIdleTimeTriggerTimer(
+        event: vscode.TextDocumentChangeEvent,
+        editor: vscode.TextEditor,
+        client: DefaultCodeWhispererClient,
+        config: ConfigurationEntry
+    ) {
+        if (this.idleTriggerTimer) {
+            clearInterval(this.idleTriggerTimer)
+            this.idleTriggerTimer = undefined
+        }
+        this.idleTriggerTimer = setInterval(() => {
+            const duration = Math.floor((performance.now() - RecommendationHandler.instance.lastInvocationTime) / 1000)
+            if (duration < CodeWhispererConstants.invocationTimeIntervalThreshold) return
+            if (isCloud9() && RecommendationHandler.instance.isGenerateRecommendationInProgress) return
+            if (isInlineCompletionEnabled() && InlineCompletionService.instance.isPaginationRunning()) return
+            if (InlineCompletion.instance.getIsActive || InlineCompletion.instance.isPaginationRunning()) return
+            this.invokeAutomatedTrigger('IdleTime', editor, client, config)
+            if (this.idleTriggerTimer) clearInterval(this.idleTriggerTimer)
+            this.idleTriggerTimer = undefined
```

Comment:
If you want to be extra defensive:
```suggestion
            try {
                this.invokeAutomatedTrigger('IdleTime', editor, client, config)
            } finally {
                 if (this.idleTriggerTimer) {
                     clearInterval(this.idleTriggerTimer)
                     this.idleTriggerTimer = undefined
                 }
            }
            })
```